core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -543,7 +560,14 @@ class ReplicaManager(val config: KafkaConfig,
           val partitionOpt = getPartition(topicPartition)
           val info = partitionOpt match {
             case Some(partition) =>
-              partition.appendRecordsToLeader(records, isFromClient, requiredAcks)
+              try {
+                partition.appendRecordsToLeader(records, isFromClient, requiredAcks)
+              } catch {
+                case e: KafkaStorageException =>
```

Comment:
It is possible to catch IOException at the the direct thrown and convert them to KafkaStorageException. But the code change required for those try/catch/indentation is probably much more than the code needed for the extra exception in the catch. 

Also, it is probably easier to verify that we catch both IOException and KafkaStrageException in the request handling code than making sure we have a try/catch around every code that touches IO. Does this make sense?